DesignSafe-CI/portal
# DES-1690, DES-1395: Update Django to 1.11 LTS; Resolve Frontend Security Vulnerabilities

## Overview: ##
This PR primarily addresses a number of security concerns both in the frontend and the backend via updating packages and removing unused package cruft.

Also removes extraneous committed packages from `designsafe/static`.


## PR Status: ##


## Related Jira tickets: ##

* [DES-1690](https://jira.tacc.utexas.edu/browse/DES-1690)
* [DES-1395](https://jira.tacc.utexas.edu/browse/DES-1395)

## Summary of Changes: ##

- commit missing migrations
- update django to 1.11 lts and update related packages
- update dockerfile; only build when debug is false
- add Makefile
- remove unneeded committed packages; use min where possible
- secure translations
- update packages to address security vulnerabilities; linting and webpack config

## Testing Steps: ##
1. run `make build` from the root project directory
2. run `make start`
3. run `npm ci`
4. run `npm run dev`
5. Ensure that all components render properly and all actions execute correctly.
